storage/engine/rocksdb/db.cc

```diff
@@ -159,6 +160,38 @@ rocksdb::Slice ToSlice(DBString s) {
   return rocksdb::Slice(s.data, s.len);
 }
 
+std::string EncodeKey(DBKey k) {
+  std::string s;
```

Comment:
`s.reserve(something)`? How concerned do we need to be about unnecessary allocations on the C++ side?